### PR TITLE
Issue353 spotless auto-apply 

### PR DESCRIPTION
**Pull request by uweschaefer**

### CODE REVIEW

**Comment by uweschaefer** [comment]
@nln42 , @JoergAdler  would you mind verifying that this also works on Mac ?

way to test:

change formatting in some java file and commit (should refuse)
set SPOTLESS_AUTO_APPLY=1 and commit (should apply change and report, on first - pass on second commit)

**Comment by nln42** [comment]
@uweschaefer : seems to need further improvement, does not work as expected

going to investigate on this issue

**Comment by uweschaefer** [comment]
> @uweschaefer : seems to need further improvement, does not work as expected
> going to investigate on this issue

@nln42  kool, thx. let me know if i can help.

File: .githooks/pre-commit.d/spotless [hunk 1]
```diff
@@ -4,23 +4,44 @@ set -e
 
 PREFIX="> pre-commit [$(basename $0)]\t\t"
 MAVEN_OPTS="-XX:+TieredCompilation -XX:TieredStopAtLevel=1 -Xms768M"
+GREP_OPTIONS=
+
+function gitDiff()
```
Comment (uweschaefer) [inline .githooks/pre-commit.d/spotless]: very nice. i always forget people might not have basic tools like these installed

**Pull request merged**